src/gennylib/src/PoolFactory.cpp

```diff
@@ -239,7 +239,12 @@ mongocxx::options::pool PoolFactory::makeOptions() const {
 
 std::unique_ptr<mongocxx::pool> PoolFactory::makePool() const {
     auto uriStr = makeUri();
-    BOOST_LOG_TRIVIAL(info) << "Constructing pool with MongoURI '" << uriStr << "'";
+
+    std::string password{*_config->get(OptionType::kAccessOption, "Password")};
+    _config->set(OptionType::kAccessOption, "Password", "[REDACTED]");
+    auto redactedUriStr = makeUri();
```

Comment:
Given that we have access to the password (`password`) wouldn't it be more efficient to substitute the string directly, rather than constructing the DSN twice?

Reply:
I think either way, we would need 2 copies of the DSN (1 actual and 1 redacted), unless we want to mutate a single copy of DSN twice (from actual to redacted then back to actual). Also, an efficient mutation would require substituting the password with a string of the same length (hence leaking the length of the password to the log), unless we want to shuffle all the characters after the password forward and then backward.

Reply:
Agreed, I just figured copying the string after generation was simpler (in terms of raw CPU and mental overhead) than generating it twice. 

In this case, I think we need to confirm that this function (`makePool`) does not ever run concurrently with any other function in this class on the same instantiation, given it is modifying `_config`. Trying to run `makePool` from two different threads, for example, would possibly generate a result where one of the pools is trying to use `Redacted` as its password.

I'd have to properly set up my IDE for this project (I am doing so right now) but I suspect that `_config` is actually a pointer to a global object, so modifying `Password` is modifying it for all users. This'd make it risky if two pools are being generated at the same time.

It's been a while since I've actually coded C++ so I might have the wrong end of the stick here, but copying a stack-local string and substituting it seems less prone-to-failure?

Reply:
Ah I see what you mean now. Yes creating the redacted copy from the actual DSN would be best then. I'll do this, thank you for the suggestion!